examples/datastreams/main.go

```diff
@@ -59,67 +81,54 @@ func main() {
 		receiverRoom.Disconnect()
 	}()
 
-	receiverRoom.RegisterTextStreamHandler("text-receive-iter", func(reader *lksdk.TextStreamReader, participantIdentity string) {
-		logger.Infow("received text stream", "participant", participantIdentity)
-		all := reader.ReadAll()
-		logger.Infow("received text", "text", all)
+	// reading text in chunks
+	receiverRoom.RegisterTextStreamHandler("text-read-iter", func(reader *lksdk.TextStreamReader, participantIdentity string) {
+		logger.Infow("received text-read-iter text stream")
+		res := ""
+		for {
+			word, err := reader.ReadString(' ')
+			logger.Infow("read word", "word", word)
+			res += word
+			if err != nil {
+				// EOF represents the end of the stream
+				if err == io.EOF {
+					break
+				} else if err == lksdk.EAGAIN {
```

Comment:
Are you leaving this to the user? I thought we could just do it internally and externally, the user either gets data or io.EOF. WDYT?

Reply:
Understood, but feels like it can be handled internally.

> notify that to the user with ErrAgain (better to return what we have than waiting for more)
I am not understanding. I think this can return (num_bytes_read, nil) and user would take those bytes and read again.

Internally, if we get (o, io.EOF) from Go APIs before the stream end, we just loop and go back and block on read?

Reply:
So just on (0, io.EOF)?

In cases of (n, io.EOF) where n > 0, do we return ErrAgain with n? Will modify the error message to day "not enough bytes available"

Reply:
Maybe my understanding is incomplete, but this what I am thinking

user
```
for {
    if chunk, err := stream.Read(); err != nil {
      // done
      break
    } else {
       data += chunk
    }
```

internally
```
func Read() ([]byte, error) {
     buf, err := buffer.Read()
     // unclear to me how the end of stream is known, assuming that is known
     if end-of-stream {
          return buf, io.EOF
     }. else if len(buf) > 0 {
         return buf, nil
     } else {
          // len(buf) == 0 and err is probably io.EOF
          time.Sleep(10 * time.Millisecond)
     }
}
```
Effectively, `Read()` blocks till there is data or end of stream.

Reply:
Understood. Just one doubt:

The Read() method takes []byte say `p` and returns `len(p)` bytes from the buffer, if buffer has less bytes, it returns whatever is available with `err == nil` and only returns io.EOF when the buffer has nothing to read from.

But, the ReadBytes() method takes a delim byte, and returns till the delim including the delim itself. But, if delim isn't present in the buffer, it returns everything available but with io.EOF.

How should we handle this? Do we return nil in these cases (before stream closure), or do we return either or ErrAgain or io.EOF?

Reply:
Please check the latest commit, this should be handled properly now!